### PR TITLE
perf(chainAdapters): replace caip properties with their high-level counterparts

### PR DESCRIPTION
**Pull request by 0xApotheosis**
Updates `@shapeshiftoss/chainAdapters` to support breaking changes introduced in https://github.com/shapeshift/lib/pull/604.

Note, requires https://github.com/shapeshift/lib/pull/604 to be published, and assumes the new version number will be `4.0.0`.

Contributes to https://github.com/shapeshift/unchained/issues/398.

### CODE REVIEW

**Review by gomesalexandre — DISMISSED** [review]
Grepped for `caip` and `getCaip` in `packages/chain-adapters` and the only references I could find are unrelated.

**Comment by gomesalexandre** [comment]
> ```
> ./src/bitcoin/BitcoinChainAdapter.test.ts:        caip19: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
> ./src/ethereum/EthereumChainAdapter.test.ts:          caip19: 'eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d',
> ./src/ethereum/EthereumChainAdapter.ts:            caip19: caip19.toCAIP19({
> ./src/cosmossdk/CosmosSdkBaseAdapter.ts:              caip19: transfer.caip19,
> ./src/cosmossdk/CosmosSdkBaseAdapter.ts:            caip19: transfer.caip19,
> ```

The `caip19:` occurences are from `chainAdapters.Account<ChainTypes>` type, where having both caip19 and `assetId` there is right.

**Comment by 0xApotheosis** [comment]
> > ```
> > ./src/bitcoin/BitcoinChainAdapter.test.ts:        caip19: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
> > ./src/ethereum/EthereumChainAdapter.test.ts:          caip19: 'eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d',
> > ./src/ethereum/EthereumChainAdapter.ts:            caip19: caip19.toCAIP19({
> > ./src/cosmossdk/CosmosSdkBaseAdapter.ts:              caip19: transfer.caip19,
> > ./src/cosmossdk/CosmosSdkBaseAdapter.ts:            caip19: transfer.caip19,
> > ```
> 
> The `caip19:` occurences are from `chainAdapters.Account<ChainTypes>` type, where having both caip19 and `assetId` there is right.

Yesser - a breaking change release of `chain-adapters` will come along that removes that. 

Future, smaller PRs 🚀

**Comment by kaladinlight** [comment]
> > ```
> > ./src/bitcoin/BitcoinChainAdapter.test.ts:        caip19: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
> > ./src/ethereum/EthereumChainAdapter.test.ts:          caip19: 'eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d',
> > ./src/ethereum/EthereumChainAdapter.ts:            caip19: caip19.toCAIP19({
> > ./src/cosmossdk/CosmosSdkBaseAdapter.ts:              caip19: transfer.caip19,
> > ./src/cosmossdk/CosmosSdkBaseAdapter.ts:            caip19: transfer.caip19,
> > ```
> 
> The `caip19:` occurences are from `chainAdapters.Account<ChainTypes>` type, where having both caip19 and `assetId` there is right.

All references of `caip2` and `caip19` will have been removed in `@shapeshiftoss/types:v4.0.0` no?

**Comment by kaladinlight** [comment]
> > > ```
> > > ./src/bitcoin/BitcoinChainAdapter.test.ts:        caip19: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
> > > ./src/ethereum/EthereumChainAdapter.test.ts:          caip19: 'eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d',
> > > ./src/ethereum/EthereumChainAdapter.ts:            caip19: caip19.toCAIP19({
> > > ./src/cosmossdk/CosmosSdkBaseAdapter.ts:              caip19: transfer.caip19,
> > > ./src/cosmossdk/CosmosSdkBaseAdapter.ts:            caip19: transfer.caip19,
> > > ```
> > 
> > 
> > The `caip19:` occurences are from `chainAdapters.Account<ChainTypes>` type, where having both caip19 and `assetId` there is right.
> 
> Yesser - a breaking change release of `chain-adapters` will come along that removes that.
> 
> Future, smaller PRs rocket

We removed from `Account` but not the `AssetBalance`? Wouldn't removal from account be breaking?

**Comment by 0xApotheosis** [comment]
> We removed from `Account` but not the `AssetBalance`? Wouldn't removal from account be breaking?

Oh hang yeah these types are coming from the types package, not the chain-adapters package - they gotta go.

Incoming commit.

**Comment by 0xApotheosis** [comment]
> We removed from `Account` but not the `AssetBalance`? Wouldn't removal from account be breaking?

@kaladinlight I double checked and both `Account` and `AssetBalance` were removed in https://github.com/shapeshift/lib/pull/604.

In the notes above, e.g. 

```
./src/cosmossdk/CosmosSdkBaseAdapter.ts:            caip19: transfer.caip19,
```

The `Transfer` type comes from `unchained`, not `lib` (`import * as unchained from '@shapeshiftoss/unchained-client'`). 

So those instances will need to stay until we update `unchained`.

**Comment by kaladinlight** [comment]
changes in https://github.com/shapeshift/lib/pull/608 will be included in this pr now as swapper depends on chain-adapters and can't have conflicting types versions

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts [hunk 1]
```diff
@@ -347,7 +353,11 @@ export class ChainAdapter
       { topic: 'txs', addresses },
       ({ data: tx }) => {
         const transfers = tx.transfers.map<chainAdapters.TxTransfer>((transfer) => ({
-          caip19: transfer.caip19,
+<<<<<<< Updated upstream
```
Comment (elmutt) [inline packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:356]: wtf

Comment (0xApotheosis) [reply packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:356]: Stash conflict :) working on it.

**Review by kaladinlight — DISMISSED** [review]
cleaned up a couple things. looks ready to roll!

**Review by elmutt — DISMISSED** [review]

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**